desdeo_emo/selection/APD_Select_constraints.py

```diff
@@ -83,6 +83,28 @@ def do(self, pop: Population, vectors: ReferenceVectors) -> List[int]:
             sub_population_index = np.atleast_1d(
                 np.squeeze(np.where(assigned_vectors == i))
             )
+
+            # Constraint check
+            if len(sub_population_index) > 1 and pop.constraint_violation is not None:
+                constraint_following_index = []
+                violation_values = []
+                for individual in sub_population_index:
+                    violation_values.append(pop.constraint_violation[individual])
+
+                    if((pop.constraint_violation[individual] > 0).all()):
+                        constraint_following_index.append(individual)
+
+                violation_values = np.asarray(violation_values)
+                violation_values = violation_values.sum(0)
```

Comment:
Use keyword arguments instead of positional arguments to make it clear to other developers what you're doing

Reply:
Also, put this inside the if statement that comes later. This only needs to be evaluated in on out of the three possible scenarios.

Reply:
*one out of three